### PR TITLE
base64 encoding and decoding functionality

### PR DESCRIPTION
**Pull request by zouhir**
please add your review \ feedback

### CODE REVIEW

File: base64.h [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+ * @achieved:
+ *
+ * @pending:
+ *
+ * @note:
+ *
+ * @TODO: Check return types from other developers
+ *
+ */
+
```
Comment (zouhir) [inline base64.h:21]: should I add `#include base64.c` here? @Mikumiku747

File: base64.h [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+ *
+ * @description: will encode every 4 6-bit character to 3 8-Bit binary bytes
+ *
+ * @TODO: research if unsigned char is the best return value for consumers
+ *
+ */
+void decodeChunk(unsigned char in_chunk[], char *output_chunk);
+
+
+/**
```
Comment (Mikumiku747) [inline base64.h]: Basically from this point on you just cut and paste into base64.c (the file with the function names in it right now is called base64.h). The purpose of this is to distinguish interface from functionality, so if we want to reuse code in the future, changes need to be made to the program, we can rely on the functions in base64.h to just work, and similarly, if we change the functions later on in base64.c to improve the efficiency, stuff that uses the function won't even notice. It's just the way it's done for C programs with more than 1 file. 
(Although, don't forget to leave that #endif in the header file). 

 - Daniel

**Review by zouhir — COMMENTED** [review]
reflected

File: base64.h (hunk 1 above)
Comment (Mikumiku747) [reply base64.h:21]: Yep, that's correct. Looks all good, let's merge it in.

**Pull request merged**